azureDatabaseUtils/src/main/java/bio/terra/workspace/azureDatabaseUtils/database/DatabaseService.java

```diff
@@ -80,4 +90,157 @@ public void restoreNamespaceRoleAccess(String namespaceRole) {
 
     databaseDao.restoreLoginPrivileges(namespaceRole);
   }
+
+  public void pgDump(
+      String dbName,
+      String dbHost,
+      String dbPort,
+      String adminUser,
+      String blobFileName,
+      String blobContainerName,
+      String blobContainerUrlAuthenticated) {
+
+    // Grant the database role (dbName) to the landing zone identity (adminUser).
+    // In theory, we should be revoking this role after the operation is complete.
+    // We are choosing to *not* revoke this role for now, because:
+    // (1) we could run into concurrency issues if multiple users attempt to clone the same
+    // workspace at once;
+    // (2) the workspace identity can grant itself access at any time, so revoking the role
+    // doesn't protect us.
+    databaseDao.grantRole(adminUser, dbName);
+
+    List<String> commandList = generateCommandList("pg_dump", dbName, dbHost, dbPort, adminUser);
+    Map<String, String> envVars = null;
+    try {
+      envVars = Map.of("PGPASSWORD", determinePassword());
+    } catch (PSQLException e) {
+      logger.error(e.getMessage());
```

Comment:
why not just let this error bubble up? As it is, the message is logged but envVars is left null probably leading to a NPE later

Reply:
+1